pkg/build/generator/generator.go

```diff
@@ -218,9 +218,11 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 	if err != nil {
 		return nil, err
 	}
-
+	if request.Binary == nil && bc.Spec.Source.Dockerfile == nil && bc.Spec.Source.Git == nil && len(bc.Spec.Source.Images) == 0 {
```

Comment:
go figure, it turns out we do have that validation, but binary is not remaining nil because we copy the value from the buildconfig when creating the build.  will fix that.